python/lib/client/dmod/client/request_clients.py

```diff
@@ -0,0 +1,206 @@
+
+    @classmethod
+    def get_response_subtype(cls) -> Type[R]:
+        return DatasetManagementResponse
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _parse_list_of_dataset_names_from_response(self, response: DatasetManagementResponse) -> List[str]:
+        # TODO: *************************************
+        # TODO: how to parse the response for a list of dataset names?
+        pass
+
+    async def create_dataset(self, name: str, category: DataCategory) -> bool:
+        # TODO: (later) consider also adding param for data to be added
+        request = DatasetManagementMessage(action=ManagementAction.CREATE, dataset_name=name, category=category)
+        self.last_response = await self.async_make_request(request)
+        return self.last_response is not None and self.last_response.success
+
+    async def list_datasets(self, category: Optional[DataCategory] = None) -> List[str]:
+        action = ManagementAction.LIST_ALL if category is None else ManagementAction.SEARCH
+        request = DatasetManagementMessage(action=action, category=category)
+        self.last_response = await self.async_make_request(request)
+        return self._parse_list_of_dataset_names_from_response(self.last_response)
+
+    async def upload_to_dataset(self, dataset_name: str, paths: List[Path]) -> bool:
+        """
+        Upload data a dataset.
+
+        Parameters
+        ----------
+        dataset_name : str
+            The name of the dataset.
+        paths : List[Path]
+            List of one or more paths of files to upload or directories containing files to upload.
+
+        Returns
+        -------
+        bool
+            Whether uploading was successful
+        """
+        # TODO: *********************************************
+        raise NotImplementedError('Function upload_to_dataset not implemented')
+
+
+class DatasetExternalClient(DatasetClient[MaaSDatasetManagementResponse],
+                            MaasRequestClient[MaaSDatasetManagementMessage, MaaSDatasetManagementResponse]):
+    """
+    Client for authenticated communication sessions via ::class:`MaaSDatasetManagementMessage` instances.
+    """
+
+    # In particular needs - endpoint_uri: str, ssl_directory: Path
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _acquire_session_info(self, use_current_values: bool = True, force_new: bool = False):
+        """
+        Attempt to set the session information properties needed to submit a maas request.
+
+        Parameters
+        ----------
+        use_current_values : bool
+            Whether to use currently held attribute values for session details, if already not None (disregarded if
+            ``force_new`` is ``True``).
+        force_new : bool
+            Whether to force acquiring a new session, regardless of data available is available on an existing session.
```

Comment:
This sentence (at the end) is not real clear.

Reply:
For now I'd like to leave this one alone.  I need to clean up some things related to these functions, but will do that in a later branch.